src/Core/Authorization/AuthorizationResolver.cs

```diff
@@ -438,8 +438,23 @@ public IEnumerable<string> GetAllowedExposedColumns(string entityName, string ro
     }
 
     /// <summary>
-    /// Helper method to extract all claims available in the HttpContext's user object and add the claims
-    /// to the claimsInRequestContext dictionary to be used for claimType -> claim lookups.
+    /// Returns a dictionary (claim name -> claim) populated from claims within a ClaimsPrincipal's ClaimsIdentity object.
+    /// Collect all unique token claims.
+    /// Claims with a value whose data type is one of the following may result in duplicate claim types which are not supported:
+    /// 1. JSON object
+    /// 2. JSON array: e.g. roles and groups claim, which are arrays of strings and are flattened to: roles: role1, roles: role2, groups: group1, groups: group2
+    /// 3. String delimited with spaces -> e.g. scp claim, which is a space delimited string of scopes and is flattened to: scp: scope1, scp: scope2. scp: scope3
+    /// claims
+    /// Background:
+    /// Claims are name valued pairs and nothing more. https://github.com/dotnet/aspnetcore/issues/13647#issuecomment-527523224
+    /// The library that parses the jwt token into claims is the one that decides *how* to resolve the claims from the token's JSON payload.
+    /// DotNet flattens the claims into a list: https://github.com/dotnet/aspnetcore/blob/282bfc1b486ae235a3395150a8d53073a57b7f43/src/Security/Authentication/OAuth/src/JsonKeyClaimAction.cs#L39-L53
+    /// Constraints
+    /// Only ignore duplicates because to ingore specific claim types (besides the role claim) is a breaking change.
+    /// e.g. When only one scope exists in a JWT token, one scp claim will be resolved: scp: scope1
+    /// However, a delimited scp claim will result in multiple claims -> scp: scope1 , scp: scope2.
+    /// We can't know which scp the user intends to use in a database policy or session context variable, so we remove all scp claims when duplicates exist.
+    /// Because claims are resolved into a dictionary by DAB, dupes have been and continue to be unsupported.
```

Comment:
Now that we understand that inherently there could be multiple claims in the request, why do we restrict ourselves to resolve them into a only simple dictionary of <string, value> in DAB? Can't we resolve them into <string, list of values>? Accordingly, for DB policy, IIRC we add WHERE clause, the equality condition can simply be expanded to an `IN` condition, no?

Reply:
Database policies would benefit from using your suggestion on how to translate a policy like `@claims.scp eq 'myHardCodedScope'` which would translate to the following tsql IF there are >1 scopes / >1 `scp` claims:
```tsql
WHERE 'myHardCodedScope' IN ('scp1', 'openid', 'profile', 'myHardCodedScope')
```
However, as you mention, we then would need to:
- Translate OData `eq` operator in all db policies referencing claims to the tsql `IN` clause ONLY when multiple instances of a claim exist. This isn't a simple change, unfortunately.
  - As opposed to implementing OData `IN` operator so developers can write db policies to match exactly how DAB will process the policy instead of needing to remember/look up in docs that `eq` translates to `in`. 
- I haven't finalized how to proceed with how `sp_set_session_context` should handle multiple instances of a claim type because that stored procedure only allows setting unique keys. e.g. `scp` claim could only appear once. and having the value be a space delimited list defeats the purpose and require even more custom tsql code in the database to parse.
  - @abratv 's suggestion could work here: _"why not simply using "scope_0", "scope_1",... "_ with the caveat that order of scopes is not guaranteed. `scope_0` may equal `openid` in one request and may be resolved as the value `scope_1` in another request.